### PR TITLE
DAOS-2172 doc: update sec/tx/fault models

### PR DESCRIPTION
**Pull request by johannlombardi**
Update the security, transaction and fault model sections from the
the storage model doc (i.e doc/storage_model.md).
Mark agent & vos checksum sections as TODO.
Update terminology document.

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

File: doc/storage_model.md [hunk 1]
```diff
@@ -68,11 +69,13 @@ A pool is only accessible to authenticated and authorized applications. Multiple
...
 
+To access a pool, a user process should connect to this pool and pass the security checks. Once granted, a pool connection can be shared (via `local2global()` and `global2local()` operations) with any or all of its peer application processes (similar to the openg() POSIX extension). This collective connect mechanism allows to avoid metadata request storm when an massively distributed job run on the datacenter. A pool connection is then revoked when the original process that issued the connection request disconnects from the pool.
```
Comment (mchaarawi) [inline doc/storage_model.md]: - s/an/a/ massively

File: src/vos/README.md [hunk 2]
```diff
@@ -472,34 +472,13 @@ A special aggregation ULT processes aggregation, yielding frequently to avoid bl
...
-return the requested extent range to the client with its computed checksum.
-When byte array extents are aggregated, VOS individually re-computes checksum of
-all extent ranges to be merged to verify correctness, and finally computes and saves the checksum for the merged extent range.
-
-Because checksum computation involves multiple layers of stack to be in-sync, VOS plans to leverage and extend either the mchecksum library or the
-<a href="https://01.org/intel&#174;-storage-acceleration-library-open-source-version">
-Intel&#174; Storage acceleration library</a>.
+## VOS Checksum Management (TODO)
+
+One of the guarantees that VOS provides is end-to-end data integrity.  Data corruption in VOS can happen while reading or writing data due to a various reasons, including leaks or failures in persistent memory, or during data transmission through the wire. VOS supports data integrity check with checksums.
```
Comment (mchaarawi) [inline src/vos/README.md]: - due to various reason, or due to a variety of reasons/causes
- do we need check after data integrity?

File: src/vos/README.md [hunk 3]
```diff
@@ -472,34 +472,13 @@ A special aggregation ULT processes aggregation, yielding frequently to avoid bl
...
+
+The VOS API for updates and writes will require checksums as arguments from its upper layer(s). VOS requires checksum for both keys and values in case of KV objects.  VOS stores the checksum along with the data.
+
+A Lookup operation on a KV will verify the checksum by computing the checksum for the key and value. If reads in byte-arrays spans over multiple extent ranges, VOS would have to recompute the checksum at the server for each individual extent range for verifying their integrity and return the computed checksum of the entire requested extent range to the client.  In case if a read requests a partial byte array extent of an existing extent range, VOS would compute the checksum of the existing extent to verify correctness and then return the requested extent range to the client with its computed checksum.  When byte array extents are aggregated, VOS individually re-computes checksum of all extent ranges to be merged to verify correctness, and finally computes and saves the checksum for the merged extent range.
```
Comment (mchaarawi) [inline src/vos/README.md]: - s/spans/span/
- spans (remove over) multiple
- In case (remove if) a read requests a ..

File: doc/storage_model.md [hunk 4]
```diff
@@ -219,35 +181,36 @@ Pool metadata are replicated on several nodes from different high-level fault do
...
+Once detected, the faulty target or servers (effectivelly a set of targets) must be excluded from the pool map. This process is triggered either manually by the administrator or automatically. Upon exclusion, the new version of the pool map is eagerly pushed to all storage targets. At this point, the pool enters a degraded mode that might require extra processing on access (e.g. reconstructing data out of erasure code). Consequently, DAOS client and storage nodes retry RPC indefinitely until they are notified of target exclusion from the pool map. At this point, all outstanding communications with the evicted target are aborted and no further messages should be sent to the target until it is explicitly reintegrated (possibly only after maintenance action).
```
Comment (wangdi1) [inline doc/storage_model.md]: Consequently, DAOS client and storage nodes retry RPC indefinitely until they are notified of target exclusion from the pool map

Probably to say "until they find an alternative replacement target from the new pool map or fail due to no other replacements." ?

File: src/vos/README.md (hunk 2 above)
Comment (wangdi1) [inline src/vos/README.md]: s/a various reasons/various reasons/ ?

**Pull request merged**